pandas/tests/dtypes/test_inference.py

```diff
@@ -1786,34 +1847,53 @@ def test_is_scalar_builtin_nonscalars(self):
         assert not is_scalar(slice(None))
         assert not is_scalar(Ellipsis)
 
-    def test_is_scalar_numpy_array_scalars(self):
-        assert is_scalar(np.int64(1))
-        assert is_scalar(np.float64(1.0))
-        assert is_scalar(np.int32(1))
-        assert is_scalar(np.complex64(2))
-        assert is_scalar(np.object_("foobar"))
-        assert is_scalar(np.str_("foobar"))
-        assert is_scalar(np.unicode_("foobar"))
-        assert is_scalar(np.bytes_(b"foobar"))
-        assert is_scalar(np.datetime64("2014-01-01"))
-        assert is_scalar(np.timedelta64(1, "h"))
-
-    def test_is_scalar_numpy_zerodim_arrays(self):
-        for zerodim in [
+    @pytest.mark.parametrize(
+        "start",
+        (
+            np.int64(1),
+            np.float64(1.0),
+            np.int32(1),
+            np.complex64(2),
+            np.object_("foobar"),
+            np.str_("foobar"),
+            np.unicode_("foobar"),
+            np.bytes_(b"foobar"),
+            np.datetime64("2014-01-01"),
+            np.timedelta64(1, "h"),
+        ),
+    )
+    def test_is_scalar_numpy_array_scalars(self, start):
```

Comment:
this is orthogonal to the rest of the PR right?  I dont think this is actually an improvement

Reply:
Right. Within pint, there is a question if a `Quantity` that wraps an array scalar is itself a duck array-scalar or if it is a zero-dimensional duck array. The current `is_scalar` however has absolutely no concept of a duck array-scalar. This PR therefore currently only tests for duck arrays. So these tests are somewhat unrelated, but I thought increasing the test coverage in that area isn't necessarily bad. However, I have no strong feeling about this - I can remove it.

Reply:
adding extra cases is great.  i was referring to parametrizing instead of keeping it all in one test

Reply:
ah, now I understand. You are probably right of course. In particular, by parametrising, the last one `test_is_scalar_numpy_arrays`, the `np.matrix` now got created outside of the test, causing the `ignore::PendingDeprecationWarning` to be ineffective, leading to the test failure in the numpy-next environment. Good catch!